include/oneapi/dpl/pstl/hetero/dpcpp/execution_sycl_defs.h

```diff
@@ -50,14 +52,34 @@ class device_policy
     device_policy(const device_policy<OtherName>& other) : q(other.queue())
     {
     }
+
+    device_policy(const device_policy& other) : q(other.queue()) {}
+    device_policy(device_policy&& other) : q(other.queue()) {}
+    device_policy&
+    operator=(const device_policy& other)
```

Comment:
Looks like not thread-safe ```operator=``` implementation.

Reply:
Good catch. I will need to change the assignment operators to grab a lock from both mutexes.

Reply:
Done.